test/src/org/exist/xquery/XQueryTest.java

```diff
@@ -1346,6 +1346,17 @@ public void instanceOfDocumentNode() throws XMLDBException {
     }
 
     @Test
+    public void instanceOfNamespaceNode() throws XMLDBException {
+        ResourceSet result = existEmbeddedServer.executeQuery("namespace test { 'test' } instance of namespace-node()");
+        assertEquals(1,result.getSize());
+        assertEquals("true", result.getResource(0).getContent().toString());
+
+        result = existEmbeddedServer.executeQuery("<x/> instance of namespace-node()");
```

Comment:
Yes, I know that is what you are asking me to test. I looked, but I cannot see how to write an XQuery test which would exercise that code path.